Return to monke (monkey juice)

ray-boop opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
Added new banana mutation to produce monkey juice juice. Plant should represent a monkey arm.
## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
fun
Removed the smell description of some NF Drinks (by me) cause the formatting didn't fit right

## Technical details
<!-- Summary of code changes for easier review. -->
Edited: (Upstream)
- seeds.yml - Line 92
- polymorph.yml - Line 146

All other additions and changes were made in _NF.

## How to test
<!-- Describe a procedure to test this feature, along with expected output/behavior. -->
very carefully...
plant, grow, grind and or eat / drink reagent -> OOOK!!!
## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->
![dead](https://github.com/user-attachments/assets/6a436252-a39a-4aff-82d1-2616e2c65136)
![harvest](https://github.com/user-attachments/assets/61f4e8af-b179-4fcf-994f-643592fe17b8)
![produce](https://github.com/user-attachments/assets/10cd204e-d919-4da3-8106-470ef88cce1e)
![seed](https://github.com/user-attachments/assets/54e1999d-bf64-4c3d-ba29-a4a8749557b0)
![stage-1](https://github.com/user-attachments/assets/f113fa12-3ef0-484c-bae9-3bff2490b5dd)
![stage-2](https://github.com/user-attachments/assets/ccc7f00d-1625-454d-b5a2-5b8bc0dabe4b)
![stage-3](https://github.com/user-attachments/assets/f251344d-ee7e-451e-b4ec-738f76bb3ee7)
![stage-4](https://github.com/user-attachments/assets/be69df7d-457e-49a9-8c0f-2ae0e21cf15d)

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read [CONTRIBUTING.md](https://github.com/new-frontiers-14/frontier-station-14/blob/master/CONTRIBUTING.md) and and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
- [x] I have reviewed the [Ship Submission Guidelines](https://frontierstation.wiki.gg/wiki/Ship_Submission_Guidelines) if relevant.
- [x] I confirm that AI tools were not used in generating the material in this PR.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
<!--
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->
:cl:
- add: New Banana Tree Mutation for more Monkey in your life
- add: New Reagent for the Banana Tree Mutation - OOOK!!
- tweak: All humanoid species can now polymorph from weh/hew juice

ray-boop commented:
not sure if the reagent in chemical.yml is correct ^^

dvir001 commented on Resources/Prototypes/Hydroponics/seeds.yml:
```diff
@@ -89,6 +89,7 @@
     - FoodBanana
   mutationPrototypes:
     - mimana
+    - monkanana #Frontier
```
```suggestion
    - monkanana # Frontier
```

dvir001 commented on Resources/Prototypes/_NF/Entities/Objects/Consumable/Food/produce.yml:
```diff
@@ -410,3 +410,35 @@
+        Quantity: 10
+  - type: Tag
+    tags:
+      - Vegetable
```
```suggestion
      - Vegetable

```

dvir001 commented on Resources/Textures/_NF/Objects/Specific/Hydroponics/monkanana.rsi/meta.json:
```diff
@@ -0,0 +1,35 @@
+            "name": "stage-4"
+        }  
+    ]
+}
```
Spaces need to be in 4 spaces

Eg:
    "states": [
        {
            "name": "dead"

dvir001 commented:
MACHINE FIND ISSUES

ray-boop commented:
Tested, no errors popped up, no crashes. Donkeykong is with us.

whatston3 commented on Resources/Locale/en-US/_NF/flavors/flavor-profiles.ftl:
```diff
@@ -16,3 +16,4 @@ flavor-complex-tangy = tangy
 flavor-complex-spices = like spices
 flavor-complex-cinnamon = like cinnamon
 flavor-complex-antiseptic = like chemical lavender
+flavor-complex-monkana = hairy and pungent
```
```suggestion
flavor-complex-monkanana = hairy and pungent

```
Newline, consistent naming.

whatston3 commented on Resources/Locale/en-US/_NF/reagents/meta/chemicals.ftl:
```diff
@@ -1,2 +1,4 @@
 reagent-name-salicylicacid = salicylic acid
 reagent-desc-salicylicacid = A powdery substance used for dermatological treatments.
+reagent-name-juicethatmakesyouoook = juice that makes you oook
+reagent-desc-juicethatmakesyouoook = A thick brown liquid, looks like hairs are in it.
```
```suggestion
reagent-desc-juicethatmakesyouoook = A thick brown liquid, looks like hairs are in it.

```
Newline.

whatston3 commented on Resources/Prototypes/_NF/Entities/Objects/Consumable/Food/produce.yml:
```diff
@@ -410,3 +410,35 @@
+    juiceSolution:
+      reagents:
+      - ReagentId: Fiber
+        Quantity: 10
```
I'd recommend reversing the juice solution and the grind solution unless you have a good reason not to.  Juices are generally extracted through juicing.

whatston3 commented on Resources/Prototypes/_NF/Entities/Objects/Consumable/Food/produce.yml:
```diff
@@ -410,3 +410,35 @@
     output: FoodOnionRings
+
+- type: entity
+  name: monkana
+  parent: FoodProduceBase
+  id: FoodMonkana
+  description: Looks like a monkeyfied banana.
+  components:
+  - type: FlavorProfile
+    flavors:
+      - monkana
```
```suggestion
  name: monkanana
  parent: FoodProduceBase
  id: FoodMonkanana
  description: Looks like a monkeyfied banana.
  components:
  - type: FlavorProfile
    flavors:
      - monkanana
```
Consistency in naming.

whatston3 commented on Resources/Prototypes/_NF/Hydroponics/seeds.yml:
```diff
@@ -173,3 +173,32 @@
+  plantRsi: _NF/Objects/Specific/Hydroponics/monkanana.rsi
+  packetPrototype: MonkananaSeeds
+  productPrototypes:
+    - FoodMonkana
```
```suggestion
    - FoodMonkanana
```
Consistency in naming.

whatston3 commented on Resources/Prototypes/_NF/Reagents/chemicals.yml:
```diff
@@ -4,3 +4,33 @@
   color: "#EEEEEE"
+
+- type: reagent
+  id: JuiceThatMakesYouOOOK
```
```suggestion
  id: JuiceThatMakesYouOok
```
For ease of finding this stuff, avoid allcaps outside of maybe acronyms.

whatston3 commented on Resources/Prototypes/_NF/Reagents/chemicals.yml:
```diff
@@ -4,3 +4,33 @@
+- type: reagent
+  id: JuiceThatMakesYouOOOK
+  name: reagent-name-juicethatmakesyouoook
+  desc: reagent-desc-juicethatmakesyouoook
```
```suggestion
  name: reagent-name-ook
  desc: reagent-desc-ook
```
Take a page out of the juice that makes you weh and make this short and readable.

whatston3 commented on Resources/Prototypes/_NF/Entities/Objects/Consumable/Food/produce.yml:
```diff
@@ -410,3 +410,35 @@
+      food:
+        maxVol: 25
+        reagents:
+        - ReagentId: JuiceThatMakesYouOOOK
```
```suggestion
        - ReagentId: JuiceThatMakesYouOok
```
If accepting the OOOK to Ook change, here's the other use.

whatston3 commented on Resources/Prototypes/_NF/Reagents/chemicals.yml:
```diff
@@ -4,3 +4,33 @@
+  id: JuiceThatMakesYouOOOK
+  name: reagent-name-juicethatmakesyouoook
+  desc: reagent-desc-juicethatmakesyouoook
+  physicalDesc: reagent-physical-desc-thick
```
You could add a "hairy" description if you want, otherwise we _do_ have fuzzy.

whatston3 commented on Resources/Prototypes/_NF/Reagents/chemicals.yml:
```diff
@@ -4,3 +4,33 @@
+            - !type:ReagentThreshold
+              min: 10
+        - !type:AdjustReagent
+          reagent: JuiceThatMakesYouOOOK
```
```suggestion
          reagent: JuiceThatMakesYouOok
```
Consistency if you change the other OOOK instances.

whatston3 commented on Resources/Prototypes/_NF/Hydroponics/seeds.yml:
```diff
@@ -173,3 +173,32 @@
+  growthStages: 4
+  potency: 10
+  yield: 3
+  idealLight: 5
```
Shouldn't this be consistent with bananas at 9?  Surely it's a tropical fruit.
```suggestion
  idealLight: 9
```

whatston3 commented on Resources/Textures/_NF/Objects/Specific/Hydroponics/monkanana.rsi/seed.png:
Don't you have the space for a whole monkey head on there?

![image](https://github.com/user-attachments/assets/49be802b-bbf9-41b2-a36b-a5785a28162f)

whatston3 commented on Resources/Textures/_NF/Objects/Specific/Hydroponics/monkanana.rsi/produce.png:
In today's episode of "what if face", what if face?

![image](https://github.com/user-attachments/assets/d52759f0-120c-402e-b292-4ba0099fc8f0)

whatston3 replied:
Tapers into a point - less turdlike?

![image](https://github.com/user-attachments/assets/bb8fea63-0f52-4bb2-bd4e-c17d9acd36a5)

ray-boop replied:
i guess we can give it a facelift

whatston3 reviewed (commented):
Generally looks alright.

ray-boop commented:
REady to review :3

ErhardSteinhauer reviewed (approved):
ook

ErhardSteinhauer commented:
Small note though: some species don't polymorph into monkey upon reaching the threshold. The list: goblins, spiders, vulps, harpies, felinids

dvir001 commented on Resources/Prototypes/Polymorphs/polymorph.yml:
```diff
@@ -143,6 +143,7 @@
     entity: MobMonkey
     forced: true
     transferName: true
+    transferHumanoidAppearance: true #Frontier
```
```suggestion
    transferHumanoidAppearance: true # Frontier
```

whatston3 commented:
Suggestions here: https://github.com/ray-boop/frontier-station-14/compare/Return-to-monke...whatston3:frontier-station-14:2025-05-02-return-to-monke-suggestions?expand=1

1. Added an entity effect to affect only humanoids, with an optional whitelist/blacklist for species.
2. Added this effect to weh juice, hew juice, and ook juice.
3. Added a monkey entity with no inventory, removed the inventory swapping (no dropping your spaghetti because it won't fit in the monkey's inventory), swapped artifact polymorphs over to use this.
4. Swapped monkanana over to fruit from vegetable.
5. Removed the monkey cube effect (please no).

I find the name "juice that makes you oook" to be a bit uninspired when you could pick something like "chimpanzene" or some other name.

whatston3 reviewed (changes requested):
requesting changes from the suggestion above

ray-boop commented:
> Suggestions here: https://github.com/ray-boop/frontier-station-14/compare/Return-to-monke...whatston3:frontier-station-14:2025-05-02-return-to-monke-suggestions?expand=1
> 
> 1. Added an entity effect to affect only humanoids, with an optional whitelist/blacklist for species.
> 2. Added this effect to weh juice, hew juice, and ook juice - now every humanoid can weh/hew/ook.
> 3. Added a monkey entity with no inventory, removed the inventory swapping (no dropping your spaghetti because it won't fit in the monkey's inventory), swapped artifact polymorphs over to use this.
> 4. Swapped monkanana over to fruit from vegetable.
> 5. Removed the monkey cube effect (please no).
> 
> I find the name "juice that makes you oook" to be a bit uninspired when you could pick something like "chimpanzene" or some other name.

how about harambeum

whatston3 commented:
> how about harambeum

but you aren't a gorilla

ray-boop commented:
> > how about harambeum
> 
> but you aren't a gorilla

Primatum?
